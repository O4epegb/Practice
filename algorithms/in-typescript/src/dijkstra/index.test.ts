import { dijkstra } from './';

const matrix1 = [
    [Infinity, 5, 0, Infinity, Infinity, Infinity],
    [Infinity, Infinity, Infinity, 15, 20, Infinity],
    [Infinity, Infinity, Infinity, 30, 35, Infinity],
    [Infinity, Infinity, Infinity, Infinity, Infinity, 20],
    [Infinity, Infinity, Infinity, Infinity, Infinity, 10],
    [Infinity, Infinity, Infinity, Infinity, Infinity, Infinity]
];

const matrix2 = [
    [Infinity, 5, 2, Infinity, Infinity, Infinity],
    [Infinity, Infinity, Infinity, 4, 2, Infinity],
    [Infinity, 8, Infinity, 7, Infinity, Infinity],
    [Infinity, Infinity, Infinity, Infinity, 6, 3],
    [Infinity, Infinity, Infinity, Infinity, Infinity, 1],
    [Infinity, Infinity, Infinity, Infinity, Infinity, Infinity]
];

describe('Dijkstra', () => {
    it('is working correctly', () => {
        expect(dijkstra(matrix1, 0, 5)).toMatchSnapshot();
        expect(dijkstra(matrix2, 0, 5)).toMatchSnapshot();
    });
});
