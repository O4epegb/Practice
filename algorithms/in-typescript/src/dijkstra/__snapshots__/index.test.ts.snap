// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dijkstra is working correctly 1`] = `
Object {
  "distance": 35,
  "path": "0 -> 1 -> 4 -> 5",
}
`;

exports[`Dijkstra is working correctly 2`] = `
Object {
  "distance": 8,
  "path": "0 -> 1 -> 4 -> 5",
}
`;
